packages/twenty-front/src/modules/workflow/hooks/useFindWorkflowWithCurrentVersion.tsx

```diff
@@ -0,0 +1,71 @@
+import { CoreObjectNameSingular } from '@/object-metadata/types/CoreObjectNameSingular';
+import { useFindManyRecords } from '@/object-record/hooks/useFindManyRecords';
+import { useFindOneRecord } from '@/object-record/hooks/useFindOneRecord';
+import {
+  Workflow,
+  WorkflowVersion,
+  WorkflowWithCurrentVersion,
+} from '@/workflow/types/Workflow';
+import { isDefined } from 'twenty-ui';
+
+export const useFindWorkflowWithCurrentVersion = (
```

Comment:
the logic is too complex for a front hook in my opinion. I think it should be in the backend. 
You can create a resolver `packages/twenty-server/src/modules/workflow/common/workflow.resolver.ts` that contains an endpoint to get the workflowWithCurrentVersion. This hook will then just call this endpoint

Reply:
I simplified the fetching logic of the current workflow version. Now, we consider the current version to be the last created version. @thomtrp thinks it's better to keep this logic on the front end for now.